### ISSUE TITLE
Feature Request: Day of the Week and seconds

### ISSUE THREAD

**Issue by SirIntellegence** [dscheinah/pkmn-watchface#1]
I absolutely love this watch face but am missing that my previous one gave me the day of the week and seconds. You can probably fit seconds into the time box and maybe the day of the week in 3-4 characters to the left or right of the month and day (maybe as a "Status condition"?).
If others don't like it, maybe add the ability to toggle it using the watch face settings?

**Comment by dscheinah** [comment]
Thank you for the feedback. I added your feature requests to my list. 

Here is, what I think so far:
- Probably the day of the week will be shown above the health bar. Let's see if I'm able to apply an easy "status condition" style. If not, it will look like the level but be aligned to the right.
- The display of seconds needs to be configurable as I prefer to turn it off to save battery.
- If I get the configuration done, I may also add date format and display of week numbers.

**Comment by SirIntellegence** [comment]
If you are able to do the "status condition" idea, you could base each day off of a status condition so one can determine the day by how it looks, but that is probably overkill. There are 5 non-volatile status conditions as far as I can tell. Burn, Freeze, Paralysis, Poisoned/Badly Poisoned, and sleep. Don't know what you may use for the others if any.
Edit: You may also want to make the seconds display smaller than the hours and minutes like my previous watch face did. Both for easier reading and to reduce the pixel update count 
Edit 2: For reference, this is the one I was using before: http://store-beta.rebble.io/app/52cc44e045ffdd31dd000180

**Comment by dscheinah** [comment]
I got a first, very simple but working version ready. I attached a screenshot to ask for a visual review:
![pebble_screenshot_2022-03-31_20-43-32](https://user-images.githubusercontent.com/45299708/161146633-1004cd73-34e9-4506-a16c-9ad70ad5d611.png)

While experimenting I experienced, that it's not a good idea to increase the complexity. So I don't intend to add more "fanciness".

Also a short remark on the font sizes: 
Currently I use the biggest fitting built-in font for HH:MM (center aligned) and the smallest of the same family for the seconds (left aligned).
If this really helps reducing the updated pixels is out of my scope of technical knowledge. It may also be, that the pebble always needs to repaint the whole screen with each update.

**Comment by SirIntellegence** [comment]
Maybe possibly add an indicator for if the watch is in silent mode? Sorry, I just start creating a list of stuff...

**Comment by SirIntellegence** [comment]
*after the page refreshes and I see your reply* Oh. That works just fine! I love it and I understand not making it more complicated! Thank-you! Also, the smaller number for seconds is mostly for readability.
The silence thing is still an option if you want to though. You don't have to. I just find myself sometimes opening the main menu just to check if the watch is in silent mode.
Edit: I am not sure about the update pixel count either? It is just a guess.

**Comment by dscheinah** [comment]
I just released the new features. Let's see if it works well and if you like my idea for displaying an active quiet time.

**Comment by SirIntellegence** [comment]
The quiet time looks nice! I was going to see if I could implement better painting for the seconds to extend battery life, but I am not sure how to install the SDK and set up the dev environment and such. I also am not proficient in C...
Some things I have found while looking at the documentation is there is a method to paint onto a portion of a bitmap. https://developer.rebble.io/developer.pebble.com/docs/c/Graphics/Drawing_Text/index.html graphics_dra_text looks to be of interest? I don't know if you can paint straight to the screen or not. There is also graphics_draw_bitmap_in_rect, where you could possibly render to a small bitmap and then to the screen?

**Comment by dscheinah** [comment]
Yes, getting the SDK ready got complicated as the official documentation no longer works. I updated the projects README with some alternatives. You can go with Docker as I do, if you are on Linux. On Windows I would recommend VirtualBox and the prebuilt disk.

How bad is the battery impact when seconds are activated? My mobile runs on Sailfish and the settings page for the watchface does not work there - so I'm not able to activate seconds rendering for daily use.

**Comment by SirIntellegence** [comment]
I am not sure for sure. I haven't been keeping good tabs on how long my watch battery lasts I will recharge tonight and see how long it lasts without and then with the seconds.

**Comment by SirIntellegence** [comment]
So, without seconds, I got 2 days and 20 hours. With seconds I got < 50 hours. Maybe around 45? So doing some optimization might be a good idea.

**Comment by dscheinah** [comment]
Thanks for the test. I agree with your opinion on doing some optimization. Let's see what can be done. But it may take a while.

**Comment by dscheinah** [comment]
OK, I did a quick test on my first guess:
The pebble needs to redraw the whole screen with each update, even if nothing changed but the current second.

In release 2.1 I now added a cache for the "battlefield". This allows the watchface to restore everything unrelated to date and time in one (hopefully performant) step. Let's see if it helps - but this is probably the only thing I can do...

**Comment by SirIntellegence** [comment]
Ok. I will test that.

**Comment by SirIntellegence** [comment]
I got a out 47 hours that time? And I actually caught it less than 30 minutes after it died, so this one is probably more accurate... I am doing a test run with the old/other watch face to see how it compares. The Battery+ app seems to be giving a pessimistic view with that. It may be the same. I will let you know when that one is finished.

**Comment by SirIntellegence** [comment]
So... The "control" actually did worse that yours XD. It only got about 35 hours... That being said, there may be a way to save battery and show seconds, though I have no right to request it since I have asked for so much already...
I don't need to see the seconds display all of the time. If I could just toggle it on and off from the watch when I need it, that would save draws. We can't use long presses on any buttons since they are taken by shortcuts. Single presses of the right buttons are reserved. Pressing the back button lights up the watch. Mabye pressing the back button twice to toggle the seconds if the seconds option is not enabled?
As I said, you don't have to. I greatly appreciate that you have done so far.

**Comment by dscheinah** [comment]
I also did some measurements and checked the time needed to render the window (code, not hardware). I only checked with the emulator and with limited precision. Let's call the minimal precision a tick. The original version took 3 to 4 ticks and the cached version needs 2 ticks. By also caching the date and day of week it gets down to 1 tick. This will come with the next version and seems to be the absolute minimum. 
So the remaining battery seems to be drawn by the secondly wake-ups and the hardware to draw the rendered window.

Now for your idea:
Actually a watchface is not allowed to detect button "clicks". So the only input type is a accelerometer tap event. I may add an option to only update seconds for a minute or so after a tap event is detected. If used together with "render seconds" the seconds will just not update until the next event. If used without, the time will be replaced by seconds for that time.
